library/constant_time_impl.h

```diff
@@ -79,7 +71,19 @@ extern volatile mbedtls_ct_uint_t mbedtls_ct_zero;
 static inline mbedtls_ct_uint_t mbedtls_ct_compiler_opaque(mbedtls_ct_uint_t x)
 {
 #if defined(MBEDTLS_CT_ASM)
+    /* Prevent false positives from Memsan - otherwise it will report the asm as
+     * accessing secret data. */
+    TEST_CF_PUBLIC(&x, sizeof(x));
+
     asm volatile ("" : [x] "+r" (x) :);
+
+    /* Mark the return value as secret. This is needed so that code of the form:
+     *
+     * if (mbedtls_ct_compiler_opaque(secret)) { ... }
+     *
+     * will fail const-flow tests.
+     */
+    TEST_CF_SECRET(&x, sizeof(x));
```

Comment:
This also has the side-effect of marking the data as secret even when it wasn't. I think I liked the previous version (disable asm under memsan) better.